Environment for creating and visualizing HFPhase1PMTParams conditions

igv4321 opened the pull request:
Environment for creating and visualizing HFPhase1PMTParams objects which configure HF Phase 1 local reconstruction. Detailed description of the environment functionality is provided in the file "readme_HFPhase1PMTParams.txt".

silviodonato committed:
move to 6r8 L1 menu

silviodonato committed:
sync with V548

silviodonato committed:
use L1Menu6r8 in PIon menu

silviodonato committed:
sync with /dev/CMSSW_8_0_0/HLT/V551

mariadalfonso committed:
fix the right siPM paramters

mariadalfonso committed:
fixed comment as a reminder to improve and reduce CPU use

silviodonato committed:
L1Menu V7

Commit:
New PopCon handler and analyzer for dealing with B ON/OFF transitions. An example is provided for EcalADCToGeVConstant.

Commit:
New template analyzer for dealing with B ON/OFF transitions using EventSetup and PoolDBOutputService. An example is provided for EcalADCToGeVConstant.

Martin-Grunewald committed:
Repair and Fix pion txt files

Martin-Grunewald committed:
v2 frozen menu does not extract

Martin-Grunewald committed:
Update L1T menu for PIon

silviodonato committed:
conflict solved

silviodonato committed:
remove git files

silviodonato committed:
remove .orig file

silviodonato committed:
use GRun in autoCondHLT

cmsbuild commented:
A new Pull Request was created by  @igv4321 (Igor Volobouev) for CMSSW_8_1_X.

It involves the following packages:

CondTools/Hcal


@ggovi, @cmsbuild, @davidlange6 can you please review it and eventually sign? Thanks.
@apfeiffer1 this is something you requested to watch as well.
@slava77, @smuzaffar you are the release manager for this.

cms-bot commands are list here cms-sw/cmssw#13028

igv4321 commented:
This PR does not modify any existing code and, of course, is not expected to modify any physics results

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1

Tested at: 04cc1d6d9806a6540d643de581536dc5355b24eb

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16067/15488/summary.html

I found follow errors while testing this PR

**Failed tests:  RelVals  AddOn**

* **RelVals**:

When I ran the RelVals I found an error in the following worklfows:
135.4 step1
<pre>runTheMatrix-results/135.4_ZEE_13+ZEEFS_13+HARVESTUP15FS+MINIAODMCUP15FS/step1_ZEE_13+ZEEFS_13+HARVESTUP15FS+MINIAODMCUP15FS.log

</pre>

* **AddOn**:

I found errors in the following addon tests:

cmsDriver.py TTbar_13TeV_TuneCUETP8M1_cfi --conditions auto:run2_mc --fast  -n 100 --eventcontent AODSIM,DQM --relval 100000,1000 -s GEN,SIM,RECOBEFMIX,DIGI:pdigi_valid,L1,DIGI2RAW,L1Reco,RECO,EI,HLT:@relval25ns,VALIDATION           --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --datatier GEN-SIM-DIGI-RECO,DQMIO --beamspot NominalCollision2015 --era Run2_25ns --magField 38T_PostLS1 : FAILED - time: date Sun Oct  2 08:15:21 2016-date Sun Oct  2 08:14:25 2016 s - exit: 23040
cmsDriver.py TTbar_13TeV_TuneCUETP8M1_cfi --conditions auto:run2_mc --fast  -n 100 --eventcontent AODSIM,DQM --relval 100000,1000 -s GEN,SIM,RECOBEFMIX,DIGI:pdigi_valid,L1,DIGI2RAW,L1Reco,RECO,EI,HLT:@relval2016,VALIDATION  --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --datatier GEN-SIM-DIGI-RECO,DQMIO --beamspot NominalCollision2015 --era Run2_2016 --magField 38T_PostLS1 : FAILED - time: date Sun Oct  2 08:26:52 2016-date Sun Oct  2 08:14:34 2016 s - exit: 23040

igv4321 commented:
Looks like these failures should be fixed by PR #16066

Martin-Grunewald committed:
Fix committed merge conflicts

slava77 commented:
OK, if I don't, somebody should restart once an IB with this PR shows up

kpedro88 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 04cc1d6d9806a6540d643de581536dc5355b24eb
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16067/15490/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16067/15490/summary.html

Martin-Grunewald committed:
PIon to run off run2_mc_pa

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_8_1_X IBs (tests are also fine). This pull request requires discussion in the ORP meeting before it's merged. @slava77, @davidlange6, @smuzaffar

silviodonato committed:
add HLT_Ele200 Ele145 to GRun

silviodonato committed:
update L1 Menu

silviodonato committed:
partial sync with /dev/CMSSW_8_0_0/HLT/V609

silviodonato committed:
complete sync with /dev/CMSSW_8_0_0/HLT/V609

davidlange6 commented on CondTools/Hcal/interface/instantiateEosTemplates.h:
```diff
@@ -0,0 +1,74 @@
+#include <boost/archive/impl/basic_binary_oarchive.ipp>
+#include <boost/archive/impl/basic_binary_oprimitive.ipp>
+
+#if BOOST_VERSION < 104000
```
@igv4321 - please start classes with a capital letter. Otherwise I don't see a reason to protect against boost version 1.40 (2009) as we are on 1.57

davidlange6 commented on CondTools/Hcal/interface/CmdLine.h:
```diff
@@ -0,0 +1,396 @@
+// January 2016
+//=========================================================================
+
+#ifndef CMDLINE_HH_
```
please add package directory in definition

davidlange6 commented on CondTools/Hcal/interface/CmdLine.h:
```diff
@@ -0,0 +1,396 @@
+#endif
+
+// Subsequent classes will throw exceptions of the following class
+class CmdLineError
```
such generic class names should be in a name space (or be prepended with some TLA)

davidlange6 commented on CondTools/Hcal/interface/instantiateEosTemplates.h:
```diff
@@ -0,0 +1,74 @@
+#define BOOST_ARCHIVE_SERIALIZER_INCLUDED
+#endif
+
+namespace boost { namespace archive {
```
I don't think we should be adding things to the boost namespace -  is this code taken from somewhere?

silviodonato committed:
CMSHLT-1058 in GRun

silviodonato committed:
move to V4.2

cmsbuild commented:
Pull request #16067 was updated. @ggovi, @cmsbuild, @davidlange6 can you please check and sign again.

igv4321 commented:
@davidlange6 
I took into account your comments about CmdLine.h and BuildFile.xml

W.r.t. instantiateEosTemplates.h: I am, of course, not defining any new classes there. This file simply instantiates templates which are not explicitly instantiated by g++ (not sure why) for the standalone executables built in the test directory. This code comes, basically from the comments in the EOS archive header files (EOS authors obviously noted this problem as well):

https://cmssdt.cern.ch/lxr/source/CondFormats/Serialization/interface/eos/portable_iarchive.hpp#0444
https://cmssdt.cern.ch/lxr/source/CondFormats/Serialization/interface/eos/portable_oarchive.hpp#0438

Commit:
Modify ZDC DetID for the extension

silviodonato committed:
move to L1Menu V9

ggovi commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

silviodonato committed:
add HLT_L1Bptx*, remove PhysicsCirculating, remove outdated path in PRef

cmsbuild commented:
+1
Tested at: ee2b011809b63da8e3bc1664597902af5092a5bb
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16067/15591/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16067/15591/summary.html

cferraio committed:
ZDCdetID from Sunanda, zdc in hcal unpacker

Commit:
Additional material

silviodonato committed:
update PIon menu

silviodonato committed:
update makeOnlinePIon

silviodonato committed:
update to /dev/CMSSW_8_0_0/HLT/V621

kpedro88 commented:
@ggovi @davidlange6 ping for pre13

Martin-Grunewald committed:
Fix EndPath removal (PIon)

kpedro88 commented:
@ggovi @davidlange6 another ping

cferraio committed:
updated topology and raw2digi, passes run the matrix

cferraio committed:
no couts

kpedro88 commented:
@ggovi @davidlange6 one more ping

ggovi commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_8_1_X IBs (tests are also fine). This pull request requires discussion in the ORP meeting before it's merged. @slava77, @davidlange6, @smuzaffar

davidlange6 commented:
I had missed the feedback from @igv4321 - thanks - however, I'm not sure why we (or rather, I'm sure we don't) need to add protection for an obsolete boost version into cmssw?

davidlange6 committed:
Merge pull request #16222 from davidlange6/xmlmem1

Fix xerces object memory leak fix in L1 mu omit

cmsbuild committed:
Merge pull request #16237 from CMS-HIN-dilepton/HLTPAMuonOfflineDQM81X

Change naming of L2/L3 DoubleMuOpen to DoubleMu0 in Offline DQM for pPb 2016

ianna committed:
Add HO, HF, and CaloTower reco geometry display

igv4321 commented:
@davidlange6 
Indeed, we don't need to add protection for an obsolete boost version. However, as I mentioned in my previous comment, I was not the original developer of this code. This means that removal of this protection needs extra work and has a chance of introducing bugs. Since further development of this code is unlikely, I don't think that potential increase in cleanliness is worth the time and effort.

kfjack committed:
Switch MuTau skim to USER data-tier

Commit:
Move the implementation of the PopCon handler for B transitions in CondTools/RunInfo: it is a tool manipulating tags based on a condition on the magnet current as seen in the RunInfo payloads.

rovere committed:
Pure reformatting for better readability

rovere committed:
Add efficiency flag to all LS-base histos

rovere committed:
Remove commented code

davidlange6 committed:
Merge pull request #15994 from makortel/cleanup2017Workflows

Clean up temporary 2017 workflows and eras, add "2017 design" workflow

cmsbuild committed:
Merge pull request #16225 from bsunanda/Run2-TB15

bsuanda:Run2-TB15 Modify the beam line for Fermilab testbeam and introduce a geometry tester

cmsbuild committed:
Merge pull request #16216 from mtosi/highPU81x

fix PU range for highPU

davidlange6 committed:
Merge pull request #16156 from matz-e/hcal-tp-2017-db-v2

Feature Bits for 2017 HCAL HF TP

cmsbuild committed:
Merge pull request #15350 from popovvp/dqm810p9

CastorBaseMonitor module removed, cleaning the rest Castor DQM modules

Commit:
Move the implementation of the EDAnalyzer for B transitions in CondTools/RunInfo: it is a tool manipulating tags based on a condition on the magnet current as seen in the RunInfo payloads.

Commit:
New geometry for HGCAL

mkirsano committed:
fix examples

ishvetso committed:
minor commit

cmsbuild committed:
Merge pull request #16129 from jfernan2/TwinMux-ThetaPlots_81X

Fixes to DT DQM to use histograms not previously declared

gartung committed:
Fix for error generating dictionary with ROOT 6.08.XX

Commit:
Make small corrections

Commit:
Define HGCal like Hcal in LPM model

Commit:
Change material name

smuzaffar committed:
Update duplicateReflexLibrarySearch.py

updated dup dict script to match TTClusterAssociationMap and TTStubAssociationMap, in TrackTriggerAssociation

smuzaffar committed:
Merge pull request #16262 from cms-sw/smuzaffar-patch-10

Update duplicateReflexLibrarySearch.py

smuzaffar committed:
Update duplicateReflexLibrarySearch.py

add mapping for (TTStub|TTCluster).*edm::refhelper::FindForDetSetVector.*Phase2TrackerDigi.* and reco::HTTTopJetProperties lost dict definition

smuzaffar committed:
Merge pull request #16264 from cms-sw/smuzaffar-patch-11

Update duplicateReflexLibrarySearch.py

makortel committed:
Add MiniAOD DQM paths back to schedule

makortel committed:
Increase PixelTripletLargeTipGenerator+MultiHitGeneratorFromChi2 maxElement to 1M

cmsbuild committed:
Merge pull request #16172 from ishvetso/PUPPI_isolation_miniAOD_electrons_photons_v2_rebased

Puppi isolation miniAOD electrons & photons v2

cmsbuild committed:
Merge pull request #16243 from intrepid42/bugfix-pseudotop-bjets-81X

Bugfix for pseudotop bjet constituents

ebrondol committed:
rpc removal to phase2_muon era

mmusich committed:
update Hcal tags for SiPM characteristics in Run1/2 MC and data and several fixes in 2017 hcal conditions

cmsbuild committed:
Merge pull request #16257 from bsunanda/Run2-TB16

bsunanda:Run2-TB16 Make small corrections

cmsbuild committed:
Merge pull request #16251 from kfjack/switch-mutauskim-to-user

Switch MuTau skim to USER data-tier

cmsbuild committed:
Merge pull request #16059 from matz-e/hcal-tp-2017-msb-db

Use database objects to set MSB in LUTs for HCAL 2017 HE TP

FHead committed:
HI High PT DQM for 81X

Commit:
allow to ask for a mother ID in MCMultiParticleFilter

ebrondol committed:
rpc removal to phase2_muon era - fix

ebrondol committed:
doPixelShapeCut off for phase2_tracker

Dr15Jones committed:
Add example for interacting with a waiting thread

The WaitingThreadIntProducer shows how one can accumulate data from
multiple streams, pass that data to an external entity (in this case
a non-framework thread), and then pass results back and continue
processing the event.

illuzate committed:
HLTriggers for UPC Analysis for 81X

cferraio committed:
addressed first round of comments, rebased

cmsbuild committed:
Merge pull request #16266 from cms-AlCaDB/Update_GlobalTags_pre13_v0

[81X] Update Hcal tags for SiPM characteristics in Run1/2 MC and data and several fixes in 2017 hcal conditions

cmsbuild committed:
Merge pull request #16255 from mkirsano/photostests

fix examples/tests in GeneratorInterface/PhotosInterface

cmsbuild committed:
Merge pull request #16254 from bsunanda/Phase2-hgx65

bsunanda:Phase2-hgx65 New geometry for HGCAL

mmusich committed:
update L1T configuration to new menu for HLT post-MD3

mmusich committed:
Add tool for dumping LS beamspot from DB

Commit:
Added functionality to TrackHistory.cc(.h) to get matched TrackingParticle

cmsbuild committed:
Merge pull request #16272 from Dr15Jones/waitingThreadExample

Add example for interacting with a waiting thread

gartung committed:
Fix for warnings from static analyzer

Geometry/VeryForwardGeometryBuilder/interface/TotemRPGeometry.h:80:5: warning: TotemRPGeometry::GetDetector is a const member function that returns a pointer or reference to a non-const object
Geometry/VeryForwardGeometryBuilder/interface/TotemRPGeometry.h:82:5: warning: TotemRPGeometry::operator[] is a const member function that returns a pointer or reference to a non-const object

gartung committed:
Update the data class names bloom filter so the pull request check
script will catch static analyzer warnings sooner.

FHead committed:
Added missing jet30 x mu monitors

gartung committed:
Fix clang static analyzer warnings:
Data Class Const Correctness	Const function returns pointer or reference to non-const member data object	/CondFormats/JetMETObjects/interface/JetResolutionObject.h	unknown	190
Data Class Const Correctness	Const function returns pointer or reference to non-const object	/CondFormats/JetMETObjects/interface/JetResolutionObject.h	getFormula	189

knoepfel committed:
Changes to avoid repeated 'hasSubProcesses' calls.

- Changed subProcesses_ to be a vector<SubProcess> object, instead of smart pointer to same.
  If empty, looping over the object is a nop.
- Change return type of popVParameterSet() to vector<ParameterSet> instead of unique_ptr to same.
  I don't believe this introduced any breaking changes, outside of FWCore.

Conflicts:
	FWCore/Framework/interface/SubProcess.h
	FWCore/Framework/src/EventProcessor.cc
	FWCore/Framework/src/SubProcess.cc
	FWCore/ParameterSet/src/ProcessDesc.cc

kpedro88 commented:
@davidlange6 can this be merged?

gartung committed:
Make a copy of formula before modifiying parameters.

smuzaffar committed:
Merge pull request #16282 from gartung/clangSA-update-data-class-filter

Update clang static analyzer data class names bloom filter

gartung committed:
Allocate copy of formula on the stack.

gartung committed:
The right way to make a copy of formula.

cmsbuild committed:
Merge pull request #16281 from gartung/Geometry-VeryForwardGeometryBuilder-clangSA-warning-fix

Fix for warnings from static analyzer for data class const correctness

cmsbuild committed:
Merge pull request #16276 from illuzate/HLTforUPCAnalysisfor81X

HLTriggers for UPC Analysis for 81X

cmsbuild committed:
Merge pull request #16235 from calabria/gemAndMe0MuonValidations81X2

Gem and me0muon validation

cmsbuild committed:
Merge pull request #16197 from igv4321/phase1-hbhe-fixes

Phase 1 HBHE fixes related to SiPM nonlinearity and cosmic run mode

cmsbuild committed:
Merge pull request #16210 from cms-tsg-storm/81XHLTAfterMD3Train

81X HLT update after-MD3 train

cmsbuild committed:
Merge pull request #16249 from ianna/full-hcal-and-calo-tower-reco

Add HO, HF, and CaloTower Reco Geometry Display

Dr15Jones committed:
Fix 'on demand' Service building

If a Service asks for another Service in its constructor and the
second Service has not yet been created, the ServiceManager will
create the second Service 'on demand'. Unfortunately, the on
demand building was not running the ParameterSet validation
code.

knoepfel committed:
Fix only spacing errors.

kpedro88 committed:
divide simhitToPhotoelectron calculation by mean crosstalk value for SiPMs

kpedro88 committed:
fix typo, add comment

cferraio committed:
fixed rpd unpacker flag

cferraio committed:
minor changes, no changes to ZDC flag

davidlange6 committed:
Merge pull request #16290 from Dr15Jones/bugfixOnDemandServiceCreation

Fix 'on demand' Service building

cmsbuild committed:
Merge pull request #16263 from rovere/Fix-AlcaBeamMonitor-LSHistograms

Fix alca beam monitor ls histograms

cmsbuild committed:
Merge pull request #16219 from makortel/customiseClusterCheckForHighPileup81x

Customise function to increase cluster number check and seeding limits (81X)

davidlange6 committed:
Merge pull request #16208 from matz-e/hcal-tp-2017-eras

Use the right era modification for 2017 HCAL TPs

illuzate committed:
HLTriggers for UPC Analysis for 81X(Fixed)

illuzate committed:
HLTriggers for UPC Analysis for 81X(Fixed)

mbandrews committed:
Revert FEStatus to run based. Get LS reports from FEMap instead.

mmusich committed:
enable fetching of geometry from DB

knoepfel committed:
Adjustments based on Chris's suggestions.

cferraio committed:
remove unpacker flag, edit customise_PPwithHI.py

mtosi committed:
fix PU binning

cmsbuild committed:
Merge pull request #16300 from illuzate/HLTforUPCAnalysisfor81X_Fixed

HLTriggers for UPC Analysis for 81X(Fixed)

Commit:
Simulation of the Fast Timing Layers

Dr15Jones committed:
Fix another place where Services created on demand

The fact that isAvailable also can trigger the creation of a
Service on demand was missed during the first round of the fix.
This now applies the same call to guarantee that fillDescriptions
gets called for all Service constructions.

Commit:
Update all analysis scripts for IsoTrack calibrations

davidlange6 committed:
Merge pull request #16311 from Dr15Jones/bugfixOnDemandServiceCreation2

Fix another place where Services created on demand

davidlange6 committed:
Merge pull request #16302 from kpedro88/FixCrosstalk

Fix crosstalk

Commit:
Fix layer=-1 issue in HE tower 17

Michael-Krohn committed:
Updated the Forward Pixel Geometry to include Flex Cables, a Carbon Fiber sheet, and extended cabling

mariadalfonso committed:
pedestalVariation in GeV instead of fC

mariadalfonso committed:
add darkCurrent noise for siPM

slava77devel committed:
no need to copy displacedVertexCandidates from the input

slava77devel committed:
simplify

rekovic committed:
Tower-counting in Layer2 and uGT - unapcerk/packer/emultaor.

Conflicts:
	EventFilter/L1TRawToDigi/src/implementations_stage2/CaloSetup.cc
	L1Trigger/L1TGlobal/data
	L1Trigger/L1TMuon/data

slava77devel committed:
limit fitVertexFromSeed to 1000 tracks in the seed; issue a warning otherwise

slava77devel committed:
limit links map reserve size

slava77devel committed:
limit links map reserve size; use unsigned int instead of 64-bit size_t

smdogra committed:
HLT Monitoring for pPb in 81X

mariadalfonso committed:
updated HBHEChannelInfo

mariadalfonso committed:
fix conflict in the HBHEPhase1Reconstructor

davidlange6 committed:
Merge pull request #16265 from makortel/fixMiniAODValidation

Fix MiniAOD validation

davidlange6 committed:
Merge pull request #16200 from delaere/othBitClusters

Propagate the overthreshold / high charge bit to phase2 clusters

davidlange6 committed:
Merge pull request #16258 from bsunanda/Run2-sim04

Run2-sim04 Update Geant4 production Physics List for HGCal

davidlange6 committed:
Merge pull request #16270 from peiffer/NewMCMultiParticleFilter81X

allow to ask for a mother ID in MCMultiParticleFilter

igv4321 commented:
@davidlange6 
If you have no further comments, please merge. The HCAL Noise Workgroup has been waiting for this code for quite some time already.

mariadalfonso committed:
fixed HBHEChannelInfo again

rekovic committed:
Remove newly added data members from CondFormats CaloParams (bools egBypassEGVetos_ and jetBypassPUS_ are currently always used and configured as false). This avoids any needs for new Global Tag.

rekovic committed:
Remove added indent and empty line to ease the CMSSW merging and possibly avoid unnecessary signatures.

davidlange6 committed:
Merge pull request #16283 from gartung/CondFormats-JetMETObjects-clangSA-warning-fix

CondFormats/JetMETObjects : Fix clang static analyzer warnings

mmusich committed:
update SiPMCharacteristics for phase-I Global Tags

Commit:
lower case for member(function)s

Commit:
lower case for member(function)s

ebrondol committed:
removing macro constant

Commit:
Merge CalibTree.hh into L3_IsoTrackCalibration.C

smdogra committed:
HLT Monitoring for pPb in 81X

smdogra committed:
HLT Monitoring for pPb in 81X

matz-e committed:
Protect against incomplete sets of dataframes.

Allows to look at the QIE10 dataframes taken in 2016.

matz-e committed:
Fix variable name.

dkotlins committed:
modify the FED IDs for phase1 in PixelUnpackingRegion

schneiml committed:
Move the disk of cables into the FPIX disks, to avoid overlaps and allow better placement.

kurtejung committed:
Fixing Gen filters to use unsmeared HepMC Product

cmsbuild committed:
Merge pull request #16312 from bsunanda/Run2-alca68

bsunanda:Run2-alca68 Update all analysis scripts for IsoTrack calibrations

cmsbuild committed:
Merge pull request #16324 from cms-l1t-offline/pr81x_noNewCondFormats_l1t-integration-towerCount_Layer2_uGT

81x L1T for HI run - tower-counting algorithm (no new CondFormats)

silviodonato committed:
Fix L1 menu used in phase1_2017 GTs

lgray committed:
make workspaces in HGCDigitizer only depend on signal size

dkotlins committed:
delete commented lines

cmsbuild committed:
Merge pull request #16326 from cms-AlCaDB/81X_updateSiPMchacterics

[81X] update SiPMCharacteristics for phase-I Global Tags

cmsbuild committed:
Merge pull request #16207 from ebrondol/phase2_fixNavigation

Phase2 OT Rings and Navigation fixed for tk geometry v4021

davidlange6 committed:
Merge pull request #16331 from kurtejung/81X_GenFilterFix

Fixing Gen filters to use unsmeared HepMC Product

mariadalfonso committed:
fix inizialization of vectors

silviodonato committed:
Rename phase1 symbolic GT

kfjack committed:
Adding special prescaler skims for high PU run

Commit:
Additional FPIX changes by Troy.

* Flex cable location and size:
z="[pixfwdDisks:DiskHalfWidth]+[pixfwd:Disk3Z]+[FlexCableLength]/2+[SpaceBetweenCableBlocksandDisk3]"
I've set  [FlexCableLength] to (642*mm-[pixfwdDisks:DiskHalfWidth]) and
[SpaceBetweenCableBlocksandDisk3] to 4mm
this appears to be giving a slight overlap at the IP end of the flex
cables (see attached overlaps file). Should I shave them down a bit?

* Carbon fiber disk location:
z = [pixfwd:Disk1Z] - [pixfwdDisks:DiskHalfWidth] - [Spacing]
Here I've currently set [Spacing] to 2mm as a place holder. I'll change
this once you report back after checking the traveller document. This
2mm placeholder appears to be definitely giving overlaps. I'll recheck
for overlaps once we have the final measure.

* Flex cable density
density  6.615->3.3075

* Felx -> Flex (files changed):
1 occurrence in pixfwdCylinder.xml
3 occurrences in pixfwdMaterials.xml
1 occurrence in pixfwdOuterDiskZminus.xml
1 occurrence in pixfwdOuterDiskZplus.xml
1 occurrence in pixfwdbladeInnerZminus.xml
1 occurrence in pixfwdbladeInnerZplus.xml
1 occurrence in pixfwdbladeOuterZminus.xml
1 occurrence in pixfwdbladeOuterZplus.xml

mmusich committed:
Update phase-I L1 menu to same as 2016 pp simulation

kfjack committed:
Adding special prescaler skims for high PU run

cmsbuild committed:
Merge pull request #16292 from knoepfel/CMSSW_8_1_X

Cleaning up subProcesses looping.

cmsbuild committed:
Merge pull request #16280 from smoortga/RetrieveTrackingParticleFromTrackHistory

Added functionality to TrackHistory.cc(.h) to get matched TrackingParticle to the Track

schneiml committed:
Grow the Service Cylinder Polycone to avoid extrusions

It now covers almost everything excluding the disks, touching disk 3 on the back and also the space between disk1 and BPIX, to fit the CF sheet there.

schneiml committed:
Marginally cut a support ring to avoid overlaps.

This overlap was caused by the slight movement of the OD rings to fit the cable disks inside the disk volumes.

schneiml committed:
Use actual size for CF sheet, after a comment from Michael Krohn

ianna committed:
Bug Fix for CutTubs xml dump

Commit:
Generator level information usage for closure tests

Commit:
Take Kevin's comment

Commit:
Take Kevin's comment

mmusich committed:
Fix Sim Geometry (payload from https://hypernews.cern.ch/HyperNews/CMS/get/calibrations/2698/1.html) and update other tags

cmsbuild committed:
Merge pull request #16347 from ianna/dd-cuttubs-bugfix

Bug Fix for CutTub xml Dump

cmsbuild committed:
Merge pull request #16339 from cms-AlCaDB/81X_L1menuForPhaseI

[81X] Update phase-I L1 menu to same as 2016 pp simulation

cmsbuild committed:
Merge pull request #16321 from smdogra/HLTpPb81X

HLT Monitoring for pPb in 81X

cmsbuild committed:
Merge pull request #16314 from bsunanda/Run2-hcx95

bsunanda:Run2-hcx95 Fix layer=-1 issue in HE tower 17

cmsbuild committed:
Merge pull request #16306 from mtosi/fixPUbinning81x

fix PU binning

cmsbuild committed:
Merge pull request #16305 from mbandrews/FixFELumi81X

Revert FEStatus to run based. Get LS reports from FEMap instead.

cmsbuild committed:
Merge pull request #16253 from diguida/BTransition_PopCon_81X_v1

New PopCon handler for dealing with B ON/OFF transitions, with test analyzer for EcalADCToGeVConstants

kpedro88 committed:
remove unnecessary Eras import

kpedro88 committed:
fix a few more Eras imports

kpedro88 committed:
fix typo

kfjack committed:
Rename to evtSplit skims

cmsbuild committed:
Merge pull request #16350 from bsunanda/Run2-alca70

bsunanda:Run2-alca70 Generator level information usage for closure tests

fioriNTU committed:
Tool for shifters

cmsbuild committed:
Merge pull request #16332 from silviodonato/patch-22

Change L1 menu for HLT tests with phase1 2017 geometry

cmsbuild committed:
Merge pull request #16143 from ianna/2017-totem

Update TOTEM in 2017 scenarios

cmsbuild committed:
Merge pull request #16074 from CTPPS/ctpps_conf_daq_mapping

CTPPS: DAQ mapping per interval-of-validity

cmsbuild committed:
Merge pull request #16098 from thomreis/l1Muon-DQM_v2_CMSSW_8_1_X

L1T muon online DQM 8_1_X - 2nd try

cmsbuild committed:
Merge pull request #16328 from matz-e/hcal-misc-fixes-v1

Small fixes for 2017 TP generation

mmusich committed:
Fix Sim Geometry (payload from https://hypernews.cern.ch/HyperNews/CMS/get/calibrations/2698/1/1.html)

dkotlins committed:
get rid of fpix bpix fed printout for regional

dkotlins committed:
get rid of fpix bpix fed printout for regional unpacking

mandrenguyen committed:
Added pA central skims

cmsbuild committed:
Merge pull request #16284 from FHead/CMSSW_8_1_X_HIHighPTDQM

HIN High-pT trigger DQM for 81X (CMSHLT-1078)

cmsbuild committed:
Merge pull request #16248 from diguida/BTransition_EDAnalyzer_81X_v0

New template one::EDAnalyzer for dealing with B ON/OFF transitions using EventSetup and PoolDBOutputService.

ianna committed:
Use Tracker Topology for Building Geometry Hierarchy

fioriNTU committed:
fixing a memory leak

cmsbuild committed:
Merge pull request #16358 from kfjack/add-highPU-special-skims

Add high pu special skims

davidlange6 committed:
Merge pull request #16150 from cferraio/rpdInUnpacker_v3

RPD implemented in 81X

kpedro88 commented:
@davidlange6 ping again

cmsbuild committed:
Merge pull request #16234 from astakia/l1t-muon-offline-dqm-810pre12

Offline DQM L1T Muon CMSSW_8_1_X

fioriNTU committed:
fix normalization of summary histo

Commit:
Added a getter for the ClusterTPAssociation::map_

schneiml committed:
Correct the Cable Disk Thickness after a comment of @krose13.

ebrondol committed:
specifying module names

cmsbuild committed:
Merge pull request #16364 from ianna/fw-reco-geometry-with-topology

Use Tracker Topology for Building Geometry Hierarchy

mmusich committed:
Update Tracker Alignment in GT to reflect geometry changes and reactivate alignment when fetching geometry from XML

Commit:
replace auto_ptr with shared_ptr

cmsbuild committed:
Merge pull request #16336 from lgray/topic_reduce_hgcdigitizer_memory

Make workspaces in HGCDigitizer only depend on signal size

davidlange6 committed:
Merge pull request #16363 from mandrenguyen/pACentralSkims_81X_v3

Added pA central skims

cmsbuild committed:
Merge pull request #15484 from ronchese/new_bph_reco_packages

new BPH reco packages

cmsbuild committed:
Merge pull request #16259 from mariadalfonso/M3disableTimeSlew

HBHE M3 and timeSlew

schneiml committed:
Add the Support Arms.

- Material is a pure guess atm.
- Z position is incorrect, to avoid overlaps.
- phi positions are roughly read from the 3d CAD, should be ok to +-1mm.

schneiml committed:
Add Cooling Tubes inside the disk volume.

- Position in X/Y is approximate, but the real tubes are expected to be randomly curved anyways (bent by hand?)
- Position in Z is chosen to avoid collisions
- Only straight parts in the X/Y plane are modeled, length sometimes limted due to collisions
- Material is a guess

fioriNTU committed:
Adding HLT paths for p-Pb collisions

slava77devel committed:
return to the old behavior of silently ignoring missing input (added LogInfo now, to be a bit more explicit)

Commit:
remove dependence of G4 in Geometry

schneiml committed:
Improve material mixtures.

These changes are *not* checked against reality. These are only guesses that should be more correct than what was there before, but all *numbers* have to be validatedagainst measurements/real material data.

fioriNTU committed:
adding back title to histograms

cmsbuild committed:
Merge pull request #16333 from dkotlins/updatePixelUnpackingRegions

modify the FED IDs for phase1 in PixelUnpackingRegion

cmsbuild committed:
Merge pull request #16195 from jshlee/GEM-HitPattern-Fix-81X

fix hitpattern for gem and me0

Commit:
Bug fix for # of bits for phi

cmsbuild committed:
Merge pull request #16367 from smoortga/GetterForTPClusterMap

Added a getter for the ClusterTPAssociation::map_

cmsbuild committed:
Merge pull request #16301 from cms-AlCaDB/81X_updateGeometryTags_phaseI_pre15

[81X] enable fetching of phase-I geometry from DB

davidlange6 committed:
Merge pull request #16181 from ebrondol/phase2Tk_eraMigration3_81X

Third part of the migration to era for phase2 tracking

mariadalfonso committed:
add the reference to the ADCquantization error for ADC10-11

igv4321 commented:
@davidlange6 
Is there something that prevents you from merging this PR?

kpedro88 committed:
remove phase2 tk customization functions

cmsbuild committed:
Merge pull request #16279 from cms-AlCaDB/81X_BeamSpotTool

Add tool for dumping LS-based beamspot from DB

cmsbuild committed:
Merge pull request #16361 from fioriNTU/ModuleOccupancyTrend81X

Adding tool to produce single module occupancy trends

davidlange6 committed:
Merge pull request #16371 from fioriNTU/DQM_fix_for_pPB_81X

Adding HLT paths for p-Pb collisions

schneiml committed:
Adjust materials according to information given by Keith Rose.

In addition  to the densities,
- all CF-Materials were unified to be the same as the FPix_Cylinder_CF
- the Support Arms got a steel component (which is currently pure Fe and not fully correct)
- The Cooling pipe  steel got 2% Mo
The cooling pipe densities were averaged, since they use the same material

schneiml committed:
Separate Inner/Outer disk cooling tube materials, and adjust densities

cmsbuild committed:
Merge pull request #16375 from slava77/CMSSW_8_1_0_pre13/sign787/PFDV-PFBA-mem

 reduction of in-module memory peaks

cmsbuild committed:
Merge pull request #16373 from ghugo83/4021_material

Material budget update for Phase 2 Tilted Trackers plugged on Phase 2 Pixel

davidlange6 committed:
Merge pull request #16357 from kpedro88/BetterEraImports2

Better era imports 2

cmsbuild committed:
Merge pull request #16309 from bsunanda/Phase2-FT07

bsunanda:Phase2-FT07 Simulation of the Fast Timing Layers

davidlange6 committed:
Merge pull request #16275 from gartung/DataFormats-HcalDetID-root60800-fix2

Fix for error generating dictionary with ROOT 6.08.XX

cmsbuild committed:
Merge pull request #15959 from mariadalfonso/siPMfixes

HBHE: siPM M2 reco fixes

kpedro88 commented:
@davidlange6 ...

davidlange6 committed:
Merge pull request #16385 from kpedro88/Phase2-WF22

remove phase2 tk customization functions

cmsbuild committed:
Merge pull request #16338 from schneiml/pixfwd-more-matbudget-changes

Updated Forward Pixel Geometry: flex cables, CF sheet, support arms, cooling tubes

kpedro88 commented:
@davidlange6 this PR was opened 29 days ago and has been fully signed for 13 days.

davidlange6 commented:
yep, maybe supporting a obsolete unusable in cms boost version in new code will seem ok tomorrow

> On Oct 30, 2016, at 7:42 PM, Kevin Pedro <notifications@github.com> wrote:
> 
> @davidlange6 this PR was opened 29 days ago and has been fully signed for 13 days.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

kpedro88 commented:
That would have been useful feedback 12 days ago

davidlange6 commented:
i think its the 3rd time i comment on the topic…

> On Oct 30, 2016, at 8:23 PM, Kevin Pedro <notifications@github.com> wrote:
> 
> That would have been useful feedback 12 days ago
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

kpedro88 commented:
Igor responded to your last comment. If his response was not acceptable, a definitive comment should have been made that the PR can't be merged in its current state. 12 days of radio silence is not productive.

davidlange6 commented:
as I said, maybe it will seem a better idea tomorrow- its clear to me that Igor believes supporting this boost version is important

> On Oct 30, 2016, at 8:59 PM, Kevin Pedro <notifications@github.com> wrote:
> 
> Igor responded to your last comment. If his response was not acceptable, a definitive comment should have been made that the PR can't be merged in its current state. 12 days of radio silence is not productive.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

igv4321 commented:
@davidlange6 @kpedro88 
David, do we have a misunderstanding? I thought I had clearly indicated in my comments the reason why support for old versions of boost was not removed. It is not because we need it, it is because I was not the original author of the code (EOS people were), and removal of this support requires some additional coding (even if this reduces to just deleting some lines) and testing effort. This effort will not improve code functionality (obviously) or even maintainability -- it is highly likely that the first version of that particular file is going to be the last one because there can be only two types of EOS archives (read, write) to take care of.

If you want us to go through some additional motions, we can do that. However, I strongly suggest that decisions of this type should be left to the people who actually have a stake in the code. This includes
code developers, maintainers, and users. At the end, this approach will simplify your own life -- you are not responsible for the CMSSW code, CMS collaboration is.

Note that, by timely resolving problems of the kind we have encountered here, you can actually provide a better service to the CMS collaboration. This, of course, implies the courtesy of replying to other people's comments within a reasonable amount of time.

Igor

davidlange6 committed:
Merge pull request #16140 from cms-met/MM_FixSchedule_230916

Fix for Puppi MET and MET significance compatibility between AOD and miniAOD

davidlange6 commented:
Hi @igv4321 - no I think we understand correctly

anyway, the good news is that CMSSW already has the code in question in CondFormats/Serialization (not sure why alca didn't note that). so it can be removed from this PR.

Commit:
Merged HFPhase1PMTParams-utilities from repository igv4321

Commit:
removing instantiateEosTemplates.h header

cmsbuild commented:
Pull request #16067 was updated. @ghellwig, @ianna, @cerminar, @fabozzi, @Martin-Grunewald, @rekovic, @vanbesien, @govoni, @perrotta, @civanch, @monttj, @silviodonato, @franzoni, @cmsbuild, @fwyzard, @davidlange6, @Dr15Jones, @cvuosalo, @mdhildreth, @slava77, @ggovi, @mmusich, @mulhearn, @perrozzi, @thuer, @dmitrijus, @srimanob, @hengne can you please check and sign again.

kpedro88 commented:
@igv4321 rebase failed (did you rebase or merge?)

igv4321 commented:
That merge didn't go well ... OK, will submit another PR

igv4321 commented:
Hi Kevin,

This PR is now resubmitted as
https://github.com/cms-sw/cmssw/pull/16403

Please start the tests.

Thanks,
Igor

On 10/31/2016 09:43 AM, Kevin Pedro wrote:
>
> @igv4321 <https://github.com/igv4321> rebase failed (did you rebase or merge?)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/cms-sw/cmssw/pull/16067#issuecomment-257312169>, or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AFS4-Wt7hX_rEn23p7vz9_6n3GzAGxBGks5q5f57gaJpZM4KL6_6>.
>
